src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -105,76 +126,105 @@ public Future<ConnectionStatus> getConnectionStatus() {
   }
 
   protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
-    var kafkaConfig = spec.kafkaClusterConfig();
-    if (kafkaConfig == null) {
+    return withAdminClient(adminClient -> {
+      // There is a Kafka configuration, so validate the connection by creating an AdminClient
+      // and describing the cluster.
+      try {
+        var clusterId = adminClient
+            .describeCluster()
+            .clusterId()
+            .get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        return Future.succeededFuture(
+            ConnectionStatusKafkaClusterStatusBuilder
+                .builder()
+                .state(ConnectedState.SUCCESS)
+                .build()
+        );
+      } catch (Exception e) {
+        // The connection failed
+        Throwable cause = e;
+        if (e instanceof java.util.concurrent.ExecutionException) {
+          cause = e.getCause();
+        }
+        // so successfully return the failed state
+        return Future.succeededFuture(
+            ConnectionStatusKafkaClusterStatusBuilder
+                .builder()
+                .state(ConnectedState.FAILED)
+                .errors(
+                    new AuthErrors().withSignIn(
+                        "Failed to connect to Kafka cluster: %s".formatted(cause.getMessage())
+                    )
+                ).build()
+        );
+      }
+    }).orElseGet(() -> {
+      // There is no Kafka cluster configuration, so return the NONE state
```

Comment:
This just refactors the `DirectConnectionState` to have a public `withAdminClient(...)` method and to use that here. The `RealDirectFetcher` can then use the `withAdminClient(...)`, avoiding some duplicate code.